hrabsence/tests/phpunit/CRM/HRAbsence/BAO/HRAbsenceTypeTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_HRAbsence_Test_Fabricator_HRAbsenceType as AbsenceTypeFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobContract as HRJobContractFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobLeave as LeaveFabricator;
+use CRM_HRCore_Test_Fabricator_Activity as ActivityFabricator;
+
+/**
+ * Class CRM_HRAbsence_BAO_HRAbsenceTypeTest
+ * Runs tests on AbsenceType BAO Class
+ * 
+ * @group headless
+ */
+class CRM_HRAbsence_BAO_HRAbsenceTypeTest extends PHPUnit_Framework_TestCase 
+  implements HeadlessInterface, TransactionalInterface {
+  
+  protected $absenceType;
+  protected $contact;
+  protected $jobContract;
+  protected $leave;
+  
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('uk.co.compucorp.civicrm.hrcore')
+      ->install('org.civicrm.hrjobcontract')
+      ->installMe(__DIR__)
+      ->apply();
+  }
+    
+  protected function setUp() {
+    // Create Absence Type
+    $this->absenceType = AbsenceTypeFabricator::fabricateUsingAPI();
+    
+    // Create Contact, Contract and Leave for Absence Type
+    $this->contact = ContactFabricator::fabricate(['first_name' => 'john', 'last_name' => 'shepard']);
+    $this->jobContract = HRJobContractFabricator::fabricate(
+      ['contact_id' => $this->contact['id']], 
+      ['period_start_date' => '2015-01-01']
+    );
+    $this->leave = LeaveFabricator::fabricate([
+      'jobcontract_id' => $this->jobContract['id'],
+      'leave_type' => $this->absenceType['id']
+    ]);
+  }
+  
+  /**
+   * Tests that absence typs used in leave requests cannot be deleted.
+   * 
+   * @expectedException CRM_Core_Exception
+   */
+  public function testDeletionOfUsedAbsenceType() {
+    // Create Request for Absence Type
+    $activityParam = [
+      'source_contact_id' => $this->contact['id'],
+      'target_contact_id' => $this->contact['id'],
+      'assignee_contact_id' => $this->contact['id'],
+      'activity_type_id' => $this->absenceType['debit_activity_type_id'],
+    ];
+    ActivityFabricator::fabricate($activityParam);
+    
+    // Try to delete, expect exception
+    CRM_HRAbsence_BAO_HRAbsenceType::del($this->absenceType['id']);
+    
+    // Verify Absence Type is still in DB
+    $result = civicrm_api3('HRAbsenceType', 'get', [

```

Comment:
If del() is expected to thrown an exception, anything after line 65 will never be executed.